### ISSUE TITLE
[Daily meeting]:  Briefing Meeting

### ISSUE THREAD

**Issue by buntynanwani** [Bootcamp-IA-P5/proyecto10-grupo5#9]
### Meeting summary

21-11-25 We Have stared working on EDA taks assiend to Oscar, Aroa, Noe and Bunty

**Comment by buntynanwani** [comment]
<html>
<body>
<html><head></head><body><h1>Toxicity Classification Project - Daily Work Report</h1>
<h2>Project Overview</h2>
<p><strong>Project Name:</strong> Intelligent Content Moderation System<br>
<strong>Team Members:</strong> Oscar Rodriguez, Bunty Nanwani, Aroa Mateo, Noé Guamán<br>
<strong>Period:</strong> November 28, 2025 - December 10, 2025<br>
<strong>Objective:</strong> Build machine learning models to classify toxic comments across 6 toxicity categories</p>
<hr>
<h2>November 28, 2025 (Thursday)</h2>
<h3>Bunty Nanwani</h3>
<ul>
<li><strong>Activity:</strong> Shared comprehensive documentation explaining overfitting reduction journey</li>
<li><strong>Achievement:</strong> Documented complete explanation of the problem, attempted solutions, and final approach</li>
<li><strong>Status:</strong> Knowledge sharing and documentation phase</li>
</ul>
<hr>
<h2>December 1, 2025 (Sunday)</h2>
<h3>Oscar Rodriguez</h3>
<ul>
<li><strong>Activity:</strong> Dataset augmentation experiments</li>
<li><strong>Approach:</strong> Abandoned translation method, switched to synonym-based augmentation</li>
<li><strong>Results Achieved:</strong>
<ul>
<li>Created dataset with 7,000 records</li>
<li><strong>Logistic Regression:</strong> Overfitting reduced to 4%</li>
<li><strong>Naive Bayes:</strong> Overfitting reduced to 2%</li>
<li>F1-Score: ~0.75-0.72 (moderate performance)</li>
</ul>
</li>
<li><strong>Dataset Shared:</strong> <code>synonym_youtoxic_english_1000.zip</code></li>
<li><strong>Target Labels (9 categories):</strong>
<ul>
<li>IsToxic, IsAbusive, IsProvocative, IsObscene, IsHatespeech, IsRacist, IsThreat, IsReligiousHate, IsNationalist</li>
</ul>
</li>
<li><strong>Key Metrics:</strong>
<ul>
<li>Logistic Regression Overfitting: 0.043023</li>
<li>Naïve Bayes Overfitting: 0.031536</li>
</ul>
</li>
<li><strong>Action:</strong> Updated backend with new models</li>
</ul>
<h3>Bunty Nanwani</h3>
<ul>
<li><strong>Activity:</strong> Started new notebook to replicate best results</li>
<li><strong>Status:</strong> Studying documentation and repeating successful training steps</li>
</ul>
<h3>Aroa Mateo</h3>
<ul>
<li><strong>Activity:</strong> Trained LightGBM model</li>
<li><strong>Results:</strong> F1-Score of 1.0000 across all 5 folds</li>
<li><strong>Issue:</strong> Metrics appeared too perfect (possible overfitting not properly detected)</li>
<li><strong>Status:</strong> Results under review</li>
</ul>
<hr>
<h2>December 4, 2025 (Wednesday)</h2>
<h3>Oscar Rodriguez</h3>
<ul>
<li><strong>Morning Activity:</strong> Shared training methodology recommendations from instructor David</li>
<li><strong>Key Recommendation:</strong> Train-Validate-Test split approach:
<ol>
<li>Split dataset: 80% training, 20% test (set aside)</li>
<li>Split training set again: 80% train, 20% validation</li>
<li>Train and optimize using train/validation sets (calculate overfitting here)</li>
<li>Final evaluation on test set (never used during training)</li>
</ol>
</li>
<li><strong>Benefit:</strong> Ensures unbiased model evaluation</li>
<li><strong>Afternoon Activity:</strong>
<ul>
<li>Created project presentation</li>
<li>Shared: <code>Intelligent Content Moderation.pptx</code></li>
<li>Updated presentation with improved results after parameter tuning</li>
</ul>
</li>
<li><strong>Deployment:</strong> Launched application to cloud
<ul>
<li><strong>URL:</strong> https://project10.osrogon.freeddns.org/</li>
</ul>
</li>
<li><strong>Pull Request:</strong> #26 - Updated LR and NB models + frontend cosmetic fixes</li>
</ul>
<h3>Bunty Nanwani</h3>
<ul>
<li><strong>Activity:</strong> Working on classical spaCy training</li>
<li><strong>Issue:</strong> AI assistant not following instructions correctly</li>
<li><strong>Status:</strong> Learning spaCy fundamentals</li>
</ul>
<h3>Aroa Mateo</h3>
<ul>
<li><strong>Issue:</strong> Pandas library errors causing problems</li>
<li><strong>Status:</strong> Troubleshooting with team support</li>
</ul>
<hr>
<h2>December 5, 2025 (Thursday)</h2>
<h3>Aroa Mateo</h3>
<ul>
<li><strong>Morning Activity:</strong> New metrics with LightGBM using NLTK</li>
<li><strong>Initial Results (9 labels):</strong>
<ul>
<li>Micro F1-Score (Optuna): 0.8546</li>
<li>Micro F1-Score (Optimized): 0.8556</li>
<li>Per-category F1 scores ranging from 0.63 to 0.97</li>
</ul>
</li>
<li><strong>Issue Identified:</strong> Still using 9 labels instead of agreed 6 labels</li>
<li><strong>Afternoon Activity:</strong> Updated to 6-label configuration</li>
<li><strong>Final Results (6 labels):</strong>
<ul>
<li>Micro F1-Score (Optuna): 0.8718</li>
<li>Micro F1-Score (Optimized): 0.8729</li>
<li>Overfitting: -0.0011 (-0.11%) - excellent generalization</li>
</ul>
</li>
<li><strong>Challenge:</strong> Overfitting calculation methodology questioned by Oscar</li>
</ul>
<h3>Oscar Rodriguez</h3>
<ul>
<li><strong>Activity:</strong> Code review and methodology guidance</li>
<li><strong>Feedback:</strong> Clarified proper overfitting calculation:
<ul>
<li>Use same model for both train and test predictions</li>
<li>Calculate: F1_train - F1_test</li>
<li>Correct usage of model.predict() on both datasets</li>
</ul>
</li>
<li><strong>Updated Results:</strong> Slightly improved after vectorizer parameter tuning</li>
<li><strong>Documentation:</strong> Updated presentation with new results</li>
</ul>
<h3>Bunty Nanwani</h3>
<ul>
<li><strong>Activity:</strong> Continued work on XGBoost model training</li>
<li><strong>Results Achieved:</strong>
<ul>
<li>TF-IDF Features: Overfitting margin 0.0466 (Excellent)</li>
<li>Word2Vec Features: Overfitting margin 0.2097 (Poor)</li>
</ul>
</li>
<li><strong>Conclusion:</strong> TF-IDF vectorization superior to Word2Vec</li>
<li><strong>Status:</strong> Comparing feature engineering approaches</li>
</ul>
<hr>
<h2>December 8, 2025 (Sunday)</h2>
<h3>Aroa Mateo</h3>
<ul>
<li><strong>Activity:</strong> Model completion and deployment preparation</li>
<li><strong>Results:</strong> Significant improvement on augmented dataset</li>
<li><strong>Deliverables:</strong>
<ul>
<li>Two models created: one with original dataset, one with augmented dataset</li>
<li>Uploaded both models for comparison</li>
</ul>
</li>
<li><strong>Status:</strong> Moving to frontend implementation</li>
</ul>
<h3>Bunty Nanwani</h3>
<ul>
<li><strong>Activity:</strong> XGBoost model deployment</li>
<li><strong>Achievement:</strong> Completed deployment of XGBoost model</li>
<li><strong>Status:</strong> Pushed to separate branch (not main) due to potential errors</li>
<li><strong>Next Steps:</strong> Team review scheduled for Monday/Wednesday</li>
</ul>
<h3>Oscar Rodriguez</h3>
<ul>
<li><strong>Activity:</strong> Reminded team to update presentation with results</li>
</ul>
<hr>
<h2>December 9, 2025 (Monday)</h2>
<h3>Bunty Nanwani</h3>
<ul>
<li><strong>Morning Activity:</strong> Presentation creation inquiry</li>
<li><strong>Tool Used:</strong> Gemini Canvas mode for presentation generation</li>
<li><strong>Question:</strong> Whether to create separate slides explaining XGBoost pipeline differences</li>
<li><strong>Documentation:</strong> Created detailed explanation slides for XGBoost methodology</li>
</ul>
<h3>Aroa Mateo</h3>
<ul>
<li><strong>Activity:</strong> Pull request created for model integration</li>
<li><strong>Status:</strong> Waiting for review and merge approval</li>
<li><strong>Next Steps:</strong> Frontend and backend integration pending</li>
</ul>
<h3>Noé Guamán</h3>
<ul>
<li><strong>Activity:</strong> Created pull request for his model</li>
<li><strong>Status:</strong> Everything functioning correctly in his branch</li>
<li><strong>Note:</strong> Waiting for Aroa's PR to be reviewed before final integration</li>
</ul>
<h3>Oscar Rodriguez</h3>
<ul>
<li><strong>Activity:</strong> Code review and architecture feedback</li>
<li><strong>Responsibilities:</strong> Reviewing pending pull requests</li>
</ul>
<hr>
<h2>December 10, 2025 (Tuesday)</h2>
<h3>Oscar Rodriguez</h3>
<ul>
<li><strong>Morning Activity:</strong> Critical bug identification in XGBoost model</li>
<li><strong>Issues Found:</strong>
<ol>
<li><strong>Same output bug:</strong> Model returning identical predictions regardless of input text</li>
<li><strong>Code duplication:</strong> Duplicate input/output schemas instead of using existing ones</li>
</ol>
</li>
<li><strong>Example Bug Evidence:</strong>
<pre><code>Text: 'string...' → IsToxic: 0.8817, IsAbusive: 0.7059Text: 'You should be banned...' → IsToxic: 0.8817, IsAbusive: 0.7059Text: 'black...' → IsToxic: 0.8817, IsAbusive: 0.7059
</code></pre></li>
<li><strong>Status:</strong> Critical issue blocking deployment</li>
</ul>
<h3>Bunty Nanwani</h3>
<ul>
<li><strong>Morning Activity:</strong> Bug investigation and troubleshooting</li>
<li><strong>Root Cause Identified:</strong> High-frequency token removal preprocessing step
<ul>
<li>Training process removed top 100 most frequent words ('black', 'people', 'white', 't', 's', 'police', etc.)</li>
<li>Deployment code wasn't applying same preprocessing</li>
<li><strong>Result:</strong> Vocabulary mismatch causing identical predictions</li>
</ul>
</li>
<li><strong>Actions Taken:</strong>
<ol>
<li>Created new branch: <code>29-fix-xgboost-model-that-gives-always-same-result</code></li>
<li>Created new notebook to address issue</li>
<li>Collaborated with instructor Yeder for support</li>
</ol>
</li>
<li><strong>Afternoon Status:</strong>
<ul>
<li>Model now giving different results</li>
<li>New issue: Model always predicting "non-toxic"</li>
<li><strong>Critical:</strong> Discussion with instructor David revealed multiple errors and duplications</li>
</ul>
</li>
<li><strong>Decision:</strong> Stopped pushing changes, keeping work in branch only</li>
<li><strong>Apology:</strong> Acknowledged uploading test files that may cause errors</li>
</ul>
<h3>Noé Guamán</h3>
<ul>
<li><strong>Morning Activity:</strong> Identified critical issue in development branch</li>
<li><strong>Issue:</strong> Development branch unexpectedly filled with numerous files</li>
<li><strong>Action:</strong> Successfully recovered clean branches</li>
<li><strong>Warning:</strong> Advised Oscar not to pull from development branch</li>
<li><strong>Status:</strong> Investigating how the problematic merge occurred</li>
</ul>
<h3>Project Status</h3>
<ul>
<li><strong>Critical Blocker:</strong> XGBoost model deployment halted due to preprocessing mismatch</li>
<li><strong>Branch Status:</strong> Development branch has integrity issues</li>
<li><strong>Team Focus:</strong> Bug fixing and branch cleanup before final delivery</li>
</ul>
<hr>
<h2>Key Achievements Summary</h2>
<h3>Overall Project Metrics</h3>
<p><strong>Final Model Performance (as of Dec 10):</strong></p>

Model | Overfitting | F1-Score | Status
-- | -- | -- | --
Logistic Regression | 4.30% | 0.75 | ✅ Deployed
Naive Bayes | 3.15% | 0.72 | ✅ Deployed
XGBoost (TF-IDF) | 4.66% | 0.82 | 🔴 Bug - Not Deployed
LightGBM | -0.11% | 0.87 | ⚠️ Under Review


<h3>Dataset Evolution</h3>
<ul>
<li><strong>Original:</strong> 800 samples → <strong>High overfitting (30%+)</strong></li>
<li><strong>Augmented v1:</strong> 3,630 samples → <strong>Improved to 4-7% overfitting</strong></li>
<li><strong>Augmented v2:</strong> 7,000 samples → <strong>Further improved to 2-4% overfitting</strong></li>
<li><strong>Final Configuration:</strong> 6 target labels (reduced from 9)</li>
</ul>
<h3>Technical Stack</h3>
<ul>
<li><strong>NLP Processing:</strong> spaCy (en_core_web_sm), NLTK</li>
<li><strong>Feature Engineering:</strong> TF-IDF, Word2Vec</li>
<li><strong>Models:</strong> Logistic Regression, Naive Bayes, XGBoost, LightGBM</li>
<li><strong>Backend:</strong> FastAPI</li>
<li><strong>Frontend:</strong> Streamlit</li>
<li><strong>Deployment:</strong> Cloud (https://project10.osrogon.freeddns.org/)</li>
</ul>
<hr>
<h2>Outstanding Issues</h2>
<h3>Critical (Blocking)</h3>
<ol>
<li><strong>XGBoost preprocessing pipeline mismatch</strong> - Causing identical predictions</li>
<li><strong>Development branch integrity</strong> - Multiple test files causing conflicts</li>
<li><strong>Model always predicting non-toxic</strong> - After attempted fix</li>
</ol>
<h3>Under Review</h3>
<ol>
<li><strong>LightGBM overfitting calculation</strong> - Methodology needs validation</li>
<li><strong>Frontend integration</strong> - Pending for Aroa and Bunty's models</li>
</ol>
<h3>Minor</h3>
<ol>
<li><strong>Code duplication</strong> - Input/output schemas need consolidation</li>
<li><strong>Documentation</strong> - Presentation needs final updates from all team members</li>
</ol>
<hr>
<h2>Team Contributions Summary</h2>
<h3>Oscar Rodriguez (Team Lead)</h3>
<ul>
<li>Dataset augmentation strategy and creation</li>
<li>Model training methodology guidance</li>
<li>Backend deployment and cloud infrastructure</li>
<li>Code reviews and architecture decisions</li>
<li>Presentation creation and management</li>
</ul>
<h3>Bunty Nanwani</h3>
<ul>
<li>XGBoost model development and optimization</li>
<li>Feature engineering comparison (TF-IDF vs Word2Vec)</li>
<li>Comprehensive documentation creation</li>
<li>Bug investigation and troubleshooting</li>
<li>Pipeline methodology research</li>
</ul>
<h3>Aroa Mateo</h3>
<ul>
<li>LightGBM model development</li>
<li>Multiple model variations (original vs augmented dataset)</li>
<li>Metric optimization and threshold tuning</li>
<li>Pull request creation for model integration</li>
</ul>
<h3>Noé Guamán</h3>
<ul>
<li>Model development and integration</li>
<li>Branch management and recovery</li>
<li>Pull request management</li>
<li>Code quality assurance</li>
</ul>
<hr>
<h2>Next Steps (Priority Order)</h2>
<ol>
<li><strong>URGENT:</strong> Fix XGBoost preprocessing pipeline to match training process</li>
<li><strong>URGENT:</strong> Clean and restore development branch integrity</li>
<li><strong>HIGH:</strong> Validate LightGBM overfitting calculations</li>
<li><strong>HIGH:</strong> Complete frontend integration for pending models</li>
<li><strong>MEDIUM:</strong> Consolidate code duplication</li>
<li><strong>MEDIUM:</strong> Final presentation updates</li>
<li><strong>LOW:</strong> Final testing and quality assurance</li>
</ol>
<hr>
<h2>Recommendations for Manager</h2>
<h3>Strengths Observed</h3>
<ul>
<li>Team collaboration and mutual support (screen sharing, code reviews)</li>
<li>Systematic approach to overfitting reduction (achieved 30% → 4%)</li>
<li>Multiple model approaches providing comparative insights</li>
<li>Good documentation and presentation preparation</li>
</ul>
<h3>Areas for Improvement</h3>
<ol>
<li><strong>Code Review Process:</strong> More rigorous review before merging to development</li>
<li><strong>Testing Strategy:</strong> Need comprehensive testing before deployment</li>
<li><strong>Branch Management:</strong> Better isolation of experimental work</li>
<li><strong>Preprocessing Consistency:</strong> Document and replicate exact training pipelines for deployment</li>
</ol>
<h3>Risk Assessment</h3>
<ul>
<li><strong>Timeline Risk:</strong> Medium - Critical bugs discovered close to deadline</li>
<li><strong>Technical Risk:</strong> Medium - XGBoost model core functionality compromised</li>
<li><strong>Quality Risk:</strong> Low - Other models performing well and deployed</li>
</ul>
<h3>Resource Needs</h3>
<ul>
<li>Additional time for thorough testing and bug fixing</li>
<li>Instructor/mentor support for complex preprocessing pipeline issues</li>
<li>Code review from external developer for quality assurance</li>
</ul></body></html>
</body>
</html>
